application-task-manager-api/src/main/java/com/xwiki/taskmanager/internal/DefaultTaskManagerConfiguration.java

```
@@ -38,21 +38,39 @@
 @Singleton
 public class DefaultTaskManagerConfiguration implements TaskManagerConfiguration
 {
+    private static final String STORAGE_FORMAT_KEY = "storageDateFormat";
+
+    private static final String DISPLAY_FORMAT_KEY = "displayDateFormat";
+
     private static final String DEFAULT_DATE_FORMAT = "yyyy/MM/dd HH:mm";
 
+    private static final String PROPERTIES_PREFIX = "taskmanager.";
+
+    @Inject
+    @Named("xwikiproperties")
+    private ConfigurationSource xwikiProperties;
+
     @Inject
     @Named("taskmanager")
     private ConfigurationSource configurationSource;
 
     @Override
     public String getStorageDateFormat()
     {
-        return this.configurationSource.getProperty("storageDateFormat", DEFAULT_DATE_FORMAT);
+        if (this.configurationSource.containsKey(STORAGE_FORMAT_KEY)) {
+            return this.configurationSource.getProperty(STORAGE_FORMAT_KEY, DEFAULT_DATE_FORMAT);
+        } else {
+            return this.xwikiProperties.getProperty(PROPERTIES_PREFIX + STORAGE_FORMAT_KEY, DEFAULT_DATE_FORMAT);
+        }
     }
 
     @Override
     public String getDisplayDateFormat()
     {
-        return this.configurationSource.getProperty("displayDateFormat", DEFAULT_DATE_FORMAT);
+        if (this.configurationSource.containsKey(DISPLAY_FORMAT_KEY)) {
+            return this.configurationSource.getProperty(DISPLAY_FORMAT_KEY, DEFAULT_DATE_FORMAT);
```

You should default on the date format configured in ``XWikiPreferences``. You can inject https://extensions.xwiki.org/xwiki/bin/view/Extension/Configuration%20Module#H22wiki22 and read the ``dateformat`` property, falling back on hard-coded date format, unfortunately (see https://github.com/xwiki/xwiki-platform/blob/xwiki-platform-14.10/xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java#L6444 ).